projects/torantie/src/halfedge.js

```diff
@@ -0,0 +1,881 @@
+// API Usage about @penrose/linear-algebra:
```

Comment:
```
/**
 * Copyright © 2021 YOUR NAME <YOUR DOMAIN>. All rights reserved.
 * Portions Copyright © 2020-2021 Changkun Ou (contact@changkun.de)
 * 
 * Use of this source code is governed by a GNU GLPv3 license that can be
 * found in the LICENSE file.
 */

```

Add the above comments to the top of every source code file that you care about your originality.
Change the following information that fits your case:

1. Add your name and your domain (if possible)
2. Remove the `Portions ...` line if a file is not derived from my solution and completely newly added